File: 1147. Longest Chunked Palindrome Decomposition/Solution.py
class Solution:
    def longestDecomposition(self, text: str) -> int:
        if not text:
            return 0
        for i in range(1, len(text) // 2 + 1):
            if text[:i] == text[-i:]:
                return 2 + self.longestDecomposition(text[i:-i])
        return 1

File: 1147. Longest Chunked Palindrome Decomposition/Solution.java
class Solution {
    public int longestDecomposition(String text) {
        if (text.isEmpty()) {
            return 0;
        }
        for (int i = 1; i <= text.length() / 2; i++) {
            if (text.substring(0, i).equals(text.substring(text.length() - i))) {
                return 2 + longestDecomposition(text.substring(i, text.length() - i));
            }
        }
        return 1;
    }
}
